mongo_orchestration/replica_sets.py

```diff
@@ -59,34 +59,61 @@ def __init__(self, rs_params):
         if not not self.sslParams:
             self.kwargs['ssl'] = True
 
-        config = {"_id": self.repl_id, "members": [
-                  self.member_create(member, index) for index, member in enumerate(rs_params.get('members', {}))
-                  ]}
+        # Always use a keyFile if auth is on.
+        if self.login and not self.auth_key:
+            self.auth_key = DEFAULT_AUTH_KEY
+
+        # Find a server that is suitable to initiate the set.
+        other_members = []
+        members = rs_params.get('members', [])
+        for index, member in enumerate(rs_params.get('members', [])):
```

Comment:
The previous code implied that 'members' is a dict, but you're returning an empty list if the key doesn't exist.